lib/vector-core/src/event/array.rs

```diff
@@ -11,29 +11,49 @@ use crate::ByteSizeOf;
 /// of events. This is effectively the same as the standard
 /// `IntoIterator<Item = Event>` implementations, but that would
 /// conflict with the base implementation for the type aliases below.
+#[allow(clippy::len_without_is_empty)]
```

Comment:
I wouldn't be mad at having an `is_empty`. We could define one by default easily enough?

Reply:
Was the reasoning here that an Event would always be of len 1?

Reply:
I skipped implementing `is_empty` because nothing uses it. Even `len` only has a single use. It is, however, trivial to implement so I'll throw it in there.